core/engine/src/value/mod.rs

```diff
@@ -351,34 +295,34 @@ impl JsValue {
     #[must_use]
     #[allow(clippy::float_cmp)]
     pub const fn as_i32(&self) -> Option<i32> {
-        match self.inner {
-            InnerValue::Integer32(integer) => Some(integer),
-            // If it can fit in a i32 and the truncated version is
-            // equal to the original then it is an integer.
-            InnerValue::Float64(rational) if rational == ((rational as i32) as f64) => {
+        if let Some(integer) = self.0.as_integer32() {
+            Some(integer)
+        } else if let Some(rational) = self.0.as_float64() {
+            // Use this poor-man's check as `[f64::fract]` isn't const.
+            if rational == ((rational as i32) as f64) {
                 Some(rational as i32)
+            } else {
+                None
             }
```

Comment:
I think we could remove this branch now, right? I ran the 262 tests locally with the f64 part removed all tests passed.